### DISCUSSION TITLE
Translations are not SSR?

### DISCUSSION THREAD

**Discussion by thomasverleye** [sergiodxa/remix-i18next#66] [Q&A] [answered]
Is it true the translations need to load client-side first before rendering?
When implementing we notice some flashes where the translation key is visible.

**Comment by sergiodxa** [comment, marked as answer]
@thomasverleye you need to load the translations on both the server and client.

On entry.server, you need to load them before rendering so the HTML comes with the correct texts.

On entry.client, you need to load them before hydrating the app because without that, the hydration will replace the translated strings with the keys and then once it loaded the translation files it will switch them back again.

**Reply by thomasverleye** [reply]
Thanks, I did something wrong in my root file :-)

**Reply by akash191095** [reply]
Hi @thomasverleye,
Can you please share how you managed to fix this?

Currently in my remix app I briefly see the translation strings and then the translations are loaded.
I tried to updated i18n and i18next config as provided in the readme but still couldn’t manage to fix this.

**Reply by thomasverleye** [reply]
hi @akash191095
Sorry for not responding that quickly, was on holiday 😎 
I fixed this by just redo-ing everything from the readme file.
Turned out I had some settings wrong from the `root.tsx` file and that I forgot to load the correct translation files in my route files (`handle` object that need to be exported).

Hopefully this could solve your problem?
If not could you share your sourcecode somewhere so we could have a look?

Cheers, Thomas

**Reply by florianchevallier** [reply]
@akash191095 did you find a way to fix this ? :D 

On my side, I always see the translations strings before seing the good ones. 

Edit : strangely enough, it seems to be fixed by removing the useSuspense: false from the i18n.ts file

**Reply by akash191095** [reply]
@florianchevallier it was an error on my part when I setup the translations
1. it was not setup correctly according to the instructions provided.
2. I was not loading the right translations needed, it was either: 
```export let handle = { i18n: "home" };```
or
```useTranslation("home")```

Sorry it was an old project and I could find it to provide more specific info

**Reply by HeroProtagonist** [reply]
I was experiencing the flash. I also missed `export let handle = { i18n: "home" };` in the component, which is in the docs. So most likely if you see the flash that is why.

**Reply by florianchevallier** [reply]
I'm not sure anymore. Do you set it for each page ? 
It's already set for the root.tsx for sure on my side

**Reply by HeroProtagonist** [reply]
I got it working. 
I don't have a handle for root, but have one for each route component that corresponds to the namespaces used by the component.
